test/helpers/kubectl.go

```diff
@@ -860,12 +860,30 @@ func (kub *Kubectl) WaitForKubeDNSEntry(serviceName, serviceNamespace string) er
 		// an IP of the pod itself, not ClusterIPNone, which is what Kubernetes
 		// shows as the IP for the service for headless services.
 		if serviceIP == v1.ClusterIPNone {
-			res := kub.ExecShort(fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP))
-			_ = kub.ExecShort(fmt.Sprintf(digCMDFallback, serviceNameWithNamespace, dnsClusterIP))
+			ress, err := kub.ExecInPods(context.TODO(), "kube-system", "k8s-app=cilium-test-logs", fmt.Sprintf(digCMD, serviceNameWithNamespace, dnsClusterIP), true)
```

Comment:
Create a context with a reasonable timeout (e.g., 1 minute) so that the test will fail if the command gets stuck.